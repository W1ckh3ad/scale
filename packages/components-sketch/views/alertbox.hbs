<div data-sketch-artboard="Alertbox" class="artboard">
  <h1 class="title--artboard">Overview</h1>


<h2 class="title--artboard-section">Colors</h2>
    <div class="agent-states--grid" data-category="Text only">
        <div class="agent-states--item">
            <p class="agent-states--label">:White</p>
            <scale-alertbox close="true" icon="true" color="white" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>
        </div>
        
        <div class="agent-states--item">
            <p class="agent-states--label">:Black</p>
<scale-alertbox close="true" iicon="true"con color="black" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">Error</p>
<scale-alertbox close="true" icon="true" color="error" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>        
      </div>
    </div>

    <h2 class="title--artboard-section">Varriant</h2>
    <div class="agent-states--grid" data-category="Text only">
        <div class="agent-states--item">
            <p class="agent-states--label">:Floating</p>
            <scale-alertbox close="true" icon="true" color="white" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>
        </div>
        
        <div class="agent-states--item">
            <p class="agent-states--label">:Outline</p>
<scale-alertbox close="true" icon="true" color="black" variant="floating">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>        </div>
    </div>





    <h2 class="title--artboard-section">Icons</h2>
    <div class="agent-states--grid" data-category="Text only">
        <div class="agent-states--item">
            <p class="agent-states--label">:Success</p>
            <scale-alertbox close="true" icon="true" color="white" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>
        </div>
        
        <div class="agent-states--item">
            <p class="agent-states--label">:Error</p>
<scale-alertbox close="true" icon="true" color="error" variant="floating">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>        </div>
    </div>

    <h2 class="title--artboard-section">Close button</h2>
    <div class="agent-states--grid" data-category="Text only">
        <div class="agent-states--item">
            <p class="agent-states--label">:Close</p>
            <scale-alertbox close="true" icon="true" color="white" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>
        </div>
        
        <div class="agent-states--item">
            <p class="agent-states--label">:No close</p>
<scale-alertbox icon="true" color="error" variant="floating">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>        </div>
    </div>

    <h2 class="title--artboard-section">Content</h2>
    <div class="agent-states--grid" data-category="Text only">
        <div class="agent-states--item">
            <p class="agent-states--label">:Only title</p>
            <scale-alertbox close="true" icon="true" color="white" variant="outline">
        <p slot="header">
          Message with title only
        </p>
      </scale-alertbox>
        </div>
        
        <div class="agent-states--item">
            <p class="agent-states--label">:Title and content</p>
<scale-alertbox color="white" variant="outline">
        <p slot="header">
          Message Title
        </p>

        <p slot="text">
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
          nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat
        </p>
      </scale-alertbox>       </div>
    </div>








  </div>
</div>


<script src="/symbol_names.js"></script>
<script>
  Array.from(document.querySelectorAll('scale-alertbox')).forEach((btn, i) => {
    btn.dataset.sketchSymbol = `Alertbox / ${order(
      1,
      getCategoryName(btn)
    )} / ${order(
      2,
      btn.dataset.hasIcon ? 'With Icon' : 'Without Icon'
    )} / ${order(4, getStateName(btn))}`;
  });
</script>
