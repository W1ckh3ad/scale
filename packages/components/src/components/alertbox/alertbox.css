:host {
  --corner-radius: 8px;
  --box-height-missing-content: 32px;
  --font-family: TeleNeo;
}

.alertbox {
  border: 1px solid var(--scl-color-white);
  border-radius: var(--corner-radius);
  width: 100%;
  font-size: var(--scl-font-size-12);
  position: relative;
}

/*Alertbox Color*/
.alertbox--color-error {
  background-color: var(--scl-color-red-70);
  color: var(--scl-color-white);
}

.alertbox.alertbox--color-black {
  background-color: var(--scl-color-black);
  color: var(--scl-color-white);
}

.alertbox.alertbox--color-white {
  background-color: var(--scl-color-white);
  color: var(--scl-color-black);
}

/*variants*/
.alertbox.alertbox--variant-outline.alertbox--color-white {
  border-radius: var(--corner-radius);
  border: solid 1px #979797;
}

.alertbox.alertbox--variant-outline.alertbox--color-black {
  border: solid 1px #979797;
  border-radius: var(--corner-radius);
}

.alertbox.alertbox--variant-floating.alertbox--color-error {
  box-shadow: 0 6px 12px 0 rgba(0, 0, 0, 0.1), 0 12px 48px 0 rgba(0, 0, 0, 0.1), inset 0 1px 0 0 #f4f4f4;
}

.alertbox.alertbox--variant-floating.alertbox--color-white {
  box-shadow: 0 6px 12px 0 rgba(0, 0, 0, 0.1), 0 12px 48px 0 rgba(0, 0, 0, 0.1), inset 0 1px 0 0 #f4f4f4;
}

.alertbox.alertbox--variant-floating.alertbox--color-black {
  box-shadow: 0 6px 12px 0 rgba(0, 0, 0, 0.1), 0 12px 48px 0 rgba(0, 0, 0, 0.1), inset 0 1px 0 0 #f4f4f4;
}


/*content*/
.alertbox .alertbox__content {
  margin-top: 8px;
  margin-left: 40px;
  margin-right: 48px;
  margin-bottom: 15px;
  font-family: var(--font-family);
  font-size: 16px;
}

/*header*/
.alertbox .alertbox__container-header {
  font-weight: bold;
  margin-left: 40px;
  margin-right: 48px;
  margin-top: 15px;
  font-family: var(--font-family);
  font-size: 16px;
}


/* Close Icon/Button */
.alertbox .alertbox__container-header .alertbox__icon-close {
  position: absolute;
  height: 15px;
  width: 15px;
  top: 13px;
  right: 16px;
}

/* Success Icon */
.alertbox .alertbox__icon-success {
  position: absolute;
  top: 13px;
  left: 17px;
  height: 10px;
  width: 14px;
}
/* Error Icon */
.alertbox .alertbox__icon-error {
  position: absolute;
  top: 13px;
  left: 17px;
  height: 10px;
  width: 14px;
}