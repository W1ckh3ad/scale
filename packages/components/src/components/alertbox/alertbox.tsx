import { Component, h, Host, Prop, Element, State } from '@stencil/core';
import classNames from 'classnames';

@Component({
  tag: 'scale-alertbox',
  styleUrl: 'alertbox.css',
  shadow: true,
})
export class Alertbox {
  @Prop() size?: 'small' | 'large' = 'small';
  @Prop() color?: 'white' | 'black' | 'error' = 'black';
  @Prop() variant?: 'floating' | 'outline';
  @Prop() icon: boolean = false;
  @Prop({ reflect: true }) close? = false;
  @State() content: boolean = true;
  @Element() element: HTMLElement;

  componentDidLoad() {
    this.content = !!this.element.querySelector("p[slot='text']");
  }

  handleIcons() {
    if (this.icon) {
      switch (this.color) {
        case 'white':
          return (
            <scale-icon-action-success
              class="alertbox__icon-success"
              accessibility-title="success"
            />
          );
        case 'black':
          return (
            <scale-icon-action-success
              class="alertbox__icon-success"
              accessibility-title="success"
            />
          );
        case 'error':
          return (
            <scale-icon-alert-error
              class="alertbox__icon-error"
              accessibility-title="error"
            />
          );
      }
    }
    return;
  }
  

  render() {
    return (
      <Host>
        <div part={this.getBasePartMap()} class={this.getCssClassMap()}>
          <div part="container" class="alertbox__container">
            {this.handleIcons()}
            <header part="header" class="alertbox__container-header">
              <slot name="header">Missing Title</slot>
              {this.close && (
                <scale-icon-action-circle-close
                  class="alertbox__icon-close"
                  onClick={() => {
                    document.querySelector('scale-alertbox').remove();
                  }}
                  accessibility-title="close"
                />
              )}
            </header>
          </div>
          {this.content && (
            <p part="content" class="alertbox__content">
              <slot name="text" />
            </p>
          )}
        </div>
      </Host>
    );
  }

  getBasePartMap() {
    return this.getCssOrBasePartMap('basePart');
  }

  getCssClassMap() {
    return this.getCssOrBasePartMap('css');
  }

  getCssOrBasePartMap(mode: 'basePart' | 'css') {
    const name = 'alertbox';
    const prefix = mode === 'basePart' ? '' : `${name}--`;

    return classNames(
      name,
      this.color && `${prefix}color-${this.color}`,
      this.variant && `${prefix}variant-${this.variant}`,
      this.icon && `${prefix}icon`
    );
  }
}
