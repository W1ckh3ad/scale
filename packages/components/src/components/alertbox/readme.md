# scale-alertbox



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type                            | Default     |
| ---------- | ---------- | ----------- | ------------------------------- | ----------- |
| `color`    | `color`    |             | `"black" \| "error" \| "white"` | `'white'`   |
| `hasclose` | `hasclose` |             | `boolean`                       | `false`     |
| `icon`     | `icon`     |             | `boolean`                       | `false`     |
| `opened`   | `opened`   |             | `boolean`                       | `undefined` |
| `timeout`  | `timeout`  |             | `boolean \| number`             | `false`     |
| `variant`  | `variant`  |             | `"floating" \| "outline"`       | `undefined` |


## Shadow Parts

| Part          | Description |
| ------------- | ----------- |
| `"container"` |             |
| `"content"`   |             |
| `"header"`    |             |


## Dependencies

### Depends on

- [scale-icon-action-success](../icons/action-success)
- [scale-icon-alert-error](../icons/alert-error)
- [scale-icon-action-circle-close](../icons/action-circle-close)

### Graph
```mermaid
graph TD;
  scale-alertbox --> scale-icon-action-success
  scale-alertbox --> scale-icon-alert-error
  scale-alertbox --> scale-icon-action-circle-close
  style scale-alertbox fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
