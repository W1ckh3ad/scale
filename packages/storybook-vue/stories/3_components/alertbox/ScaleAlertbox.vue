<template>
  <scale-alertbox
    :color="color"
    :variant="variant"
    :icon="icon"
    :hasclose="hasclose"
    :timeout="timeout"
    :opened="opened"
  >
    <slot name="header"></slot>
    <slot name="text"></slot>
  </scale-alertbox>
</template>

<script>
export default {
  props: {
    color: { type: 'white' | 'black' | 'error', default: 'white' },
    variant: { type: 'floating' | 'outline' },
    icon: { type: Boolean, default: false },
    hasclose: { type: Boolean, default: false },
    opened: { type: Boolean },
    timeout: { type: Boolean | Number, default: false },
  },
};
</script>
