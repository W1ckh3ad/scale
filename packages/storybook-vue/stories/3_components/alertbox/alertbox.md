# Alertbox

Alertboxes are contextual notifications to the user. They stand out visually from the rest of the page so that the user can recognize and read them as directly and easily as possible.

## General

### When to Use

Alertboxes are used to draw the user's attention to information specific to the current context. For example, an error when entering data into a form.

### When Not to Use

Messages that are more global in nature, i.e. not directly related to the content of the current page, should be displayed as a global notification.

## Variants

### Floating / Shadow

Use the floating variant only if the alertbox is fixed, i.e. not scrolled with the content page.

![Image Name](assets/3_components/alertbox/image-20200811081301713.png)

### Plain

Use the plain variant if the contrast of the alertbox to the background is sufficient even without an outline.

![Image Name](assets/3_components/alertbox/image-20200811081311813.png)

### Outline

Use the outline variant if the contrast of the alertbox to the background is too low.

![Image Name](assets/3_components/alertbox/image-20200811081320535.png)

## Colors

### White

![Image Name](assets/3_components/alertbox/image-20200811081338109.png)

### Black

![Image Name](assets/3_components/alertbox/image-20200811081345433.png)

### Red / Error

The red alertbox is not available as an outline version as it is not intended to be placed on a red background.

![Image Name](assets/3_components/alertbox/image-20200811081301713.png)

## Sizes

### Large

Large is the default size.

![Image Name](assets/3_components/alertbox/image-20200811081338109.png)

### Small

Due to poor legibility, use the small size in exceptional cases only.

![Image Name](assets/3_components/alertbox/image-20200811081327934.png)

## Elements

![Image Name](assets/3_components/alertbox/image-20200811081353227.png)

### Title (1)

### Icon (2) (optional)

### Close Icon (3) (optional)

### Content (4) (optional)

The content can consist of almost all types of elements e.g. links, pictures. However, keep complexity limited in order to ensure good comprehensibility.
