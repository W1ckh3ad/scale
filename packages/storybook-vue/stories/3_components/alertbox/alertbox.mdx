import {
  Meta,
  Story,
  Props,
  Preview,
  Description,
} from "@storybook/addon-docs/blocks";
import description from "./alertbox.md";
<Meta
  title="Components/Alertbox"
/>
