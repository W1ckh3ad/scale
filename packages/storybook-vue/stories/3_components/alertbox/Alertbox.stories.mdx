import {
  Meta,
  Story,
  ArgsTable,
  Canvas,
  Description,
} from '@storybook/addon-docs/blocks';
import ScaleAlertbox from './ScaleAlertbox.vue';

<Meta
  title="Beta Components/Alertbox"
  component={ScaleAlertbox}
  argTypes={{
    color: {
      defaultValue: 'black',
      options: ['white', 'black', 'error'],
      control: {
        type: 'select',
      },
    },
    variant: {
      options: ['outline', 'floating'],
      control: {
        type: 'select',
      },
    },
    icon: {
      defaultValue: 'false',
      options: ['true', 'false'],
      control: {
        type: 'select',
      },
    },
    close: {
      defaultValue: 'false',
      options: ['true', 'false'],
      control: {
        type: 'select',
      },
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  components: { ScaleAlertbox },
  props: {
    ...ScaleAlertbox.props,
    color: {
      defaultValue: 'white',
      options: ['white', 'black', 'error'],
      control: {
        type: 'select',
      },
    },
    variant: {
      options: ['outline', 'floating'],
      control: {
        type: 'select',
      },
    },
    icon: {
      defaultValue: 'true',
      options: ['true', 'false'],
      control: {
        type: 'select',
      },
    },
    close: {
      defaultValue: 'true',
      options: ['true', 'false'],
      control: {
        type: 'select',
      },
    },
  },
  template: `
      <scale-alertbox color="white" icon close variant="floating">
        <p slot="header">
          Message with title, text and closeMessage with title, text and
          closeMessage with title, text and closeMessage with title, text and
          close
        </p>
        <p slot="text">
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
          nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat
        </p>
      </scale-alertbox>
  `,
});

## Standard
### All default values

<Canvas withSource="none">
  <Story
    name="Standard"
    args={{
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />

```html
<scale-alertbox color="white" icon close variant="outline">
        <p slot="header">
          Message with title, text and closeMessage with title, text and
          closeMessage with title, text and closeMessage with title, text and
          close
        </p>
        <p slot="text">
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
          nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat
        </p>
      </scale-alertbox>
```



### Scoped CSS variables

```css
 :host {
 --corner-radius: 8px;
  --box-height-missing-content: 32px;
  --font-family: TeleNeo;
}
```


## Color
There are three colors. White, black & error.

<Canvas withSource="none">
  <Story
    name="Colors"
    args={{
      color: "error",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
 <scale-alertbox color="white" icon close variant="floating">
        <p slot="header">
          Message with title, text and closeMessage with title, text and
          closeMessage with title, text and closeMessage with title, text and
          close
        </p>
        <p slot="text">
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
          nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat
        </p>
      </scale-alertbox>
```

## Icon - Success & Error
There is a success & error icons 

<Canvas withSource="none">
  <Story
    name="Colors"
    args={{
      color: "error",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
 <scale-alertbox color="white" icon close variant="floating">
        <p slot="header">
          Message with title, text and closeMessage with title, text and
          closeMessage with title, text and closeMessage with title, text and
          close
        </p>
        <p slot="text">
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
          nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat
        </p>
     
```


## Close Button - Enalbled & Dissabled

The user has the possibility to either enable or dissable the close button

<Canvas withSource="none">
  <Story
    name="Colors"
    args={{
      color: "error",
      close: false,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
 <scale-alertbox color="white" icon variant="floating">
        <p slot="header">
          Message with title, text and closeMessage with title, text and
          closeMessage with title, text and closeMessage with title, text and
          close
        </p>
        <p slot="text">
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
          nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat
        </p>
     
```

