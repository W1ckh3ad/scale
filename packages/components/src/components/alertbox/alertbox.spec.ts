import { newSpecPage } from '@stencil/core/testing';
// import { remove } from 'lodash';
import { Alertbox } from './alertbox';


describe('Alertbox', () => {
    let element;
  
    beforeEach(async () => {
      element = new Alertbox();
    });


 it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Alertbox],
      html: `<scale-alertbox opened="true">Label</scale-alertbox>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.color = 'magenta';
    expect(element.getCssClassMap()).toContain('color-magenta');

    element.variant = 'variant';
    expect(element.getCssClassMap()).toContain('variant-variant');

    element.icon = true;
    expect(element.getCssClassMap()).toContain('icon');
  });

  /* it('should handle click remove', async () => {
    const page = await newSpecPage({
      components: [Alertbox],
      html: `
      <scale-alertbox>
      </scale-alerbox>`,
    });
    expect(page.root).toMatchSnapshot();
      const element = new Alertbox();
      element.close = false;
      document.querySelector('scale-alertbox').remove();
      expect(element).toBe(remove);
  });

    it('element is set on componentWillLoad', () => {
      const element = new Alertbox();
      element.componentDidLoad();
      expect(element.content).toBe("p[slot='text']");
    }); */

    // it('should close the alertbox', () => {
    //   expect(element.opened).toBe();
    //   element.close();
    //   expect(element.opened).toBe(false);
    // });


    it('handle color prop', async () => {
      let page = await newSpecPage({
        components: [Alertbox],
        html: `<scale-alertbox opened="true" icon="true">Label</scale-alertbox>`,
      });
      page.root.color="black";
      await page.waitForChanges();
      expect(page.root).toMatchSnapshot();
      page.root.color="blue";
      await page.waitForChanges();

      expect(page.root).toMatchSnapshot();
      page.root.color="error";
      await page.waitForChanges();
      expect(page.root).toMatchSnapshot();
      page.root.color="yellow";
      await page.waitForChanges();
      expect(page.root).toMatchSnapshot();
      page.root.color="green";
      await page.waitForChanges();
      expect(page.root).toMatchSnapshot();
      
    });

 it('should reflect attributes/props', async () => {
  const page = await newSpecPage({
    components: [Alertbox],
    html: `<scale-alertbox
                color ="black"
                variant ="outline"
                icon ="true"
                hasClose ="true"
                content ="false">
              </scale-alertbox>`,
  });
  
  expect(page.rootInstance.color).toBe('black');
  expect(page.rootInstance.variant).toBe('outline');
  expect(page.rootInstance.icon).toBe(true);
  expect(page.rootInstance.hasclose).toBe(true);
  expect(page.rootInstance.content).toBe(false);
 });
});
