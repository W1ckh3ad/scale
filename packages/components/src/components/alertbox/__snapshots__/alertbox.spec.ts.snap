// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alertbox handle color prop 1`] = `
<scale-alertbox icon="true" opened="">
  <mock:shadow-root>
    <div class="alertbox alertbox--color-black alertbox--icon" part="alertbox color-black icon">
      <div class="alertbox__container" part="container">
        <scale-icon-action-success accessibility-title="success" class="alertbox__icon-success"></scale-icon-action-success>
        <header class="alertbox__container-header" part="header">
          <slot name="header">
            <p>
              Missing Title
            </p>
          </slot>
        </header>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-alertbox>
`;

exports[`Alertbox handle color prop 2`] = `
<scale-alertbox icon="true" opened="">
  <mock:shadow-root>
    <div class="alertbox alertbox--color-blue alertbox--icon" part="alertbox color-blue icon">
      <div class="alertbox__container" part="container">
        <scale-icon-action-informational accessibility-title="info" class="alertbox__icon-info"></scale-icon-action-informational>
        <header class="alertbox__container-header" part="header">
          <slot name="header">
            <p>
              Missing Title
            </p>
          </slot>
        </header>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-alertbox>
`;

exports[`Alertbox handle color prop 3`] = `
<scale-alertbox icon="true" opened="">
  <mock:shadow-root>
    <div class="alertbox alertbox--color-error alertbox--icon" part="alertbox color-error icon">
      <div class="alertbox__container" part="container">
        <scale-icon-alert-error accessibility-title="error" class="alertbox__icon-error"></scale-icon-alert-error>
        <header class="alertbox__container-header" part="header">
          <slot name="header">
            <p>
              Missing Title
            </p>
          </slot>
        </header>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-alertbox>
`;

exports[`Alertbox handle color prop 4`] = `
<scale-alertbox icon="true" opened="">
  <mock:shadow-root>
    <div class="alertbox alertbox--color-yellow alertbox--icon" part="alertbox color-yellow icon">
      <div class="alertbox__container" part="container">
        <scale-icon-action-warning accessibility-title="warning" class="alertbox__icon-warning"></scale-icon-action-warning>
        <header class="alertbox__container-header" part="header">
          <slot name="header">
            <p>
              Missing Title
            </p>
          </slot>
        </header>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-alertbox>
`;

exports[`Alertbox handle color prop 5`] = `
<scale-alertbox icon="true" opened="">
  <mock:shadow-root>
    <div class="alertbox alertbox--color-green alertbox--icon" part="alertbox color-green icon">
      <div class="alertbox__container" part="container">
        <scale-icon-action-success accessibility-title="warsuccessning" class="alertbox__icon-success"></scale-icon-action-success>
        <header class="alertbox__container-header" part="header">
          <slot name="header">
            <p>
              Missing Title
            </p>
          </slot>
        </header>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-alertbox>
`;

exports[`Alertbox should match snapshot 1`] = `
<scale-alertbox opened="">
  <mock:shadow-root>
    <div class="alertbox alertbox--color-white" part="alertbox color-white">
      <div class="alertbox__container" part="container">
        <header class="alertbox__container-header" part="header">
          <slot name="header">
            <p>
              Missing Title
            </p>
          </slot>
        </header>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-alertbox>
`;
